src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -5998,8 +5998,8 @@ void AuraEffect::HandlePeriodicHealthLeechAuraTick(Unit* target, Unit* caster) c
 
     caster->CalcAbsorbResist(target, GetSpellInfo()->GetSchoolMask(), DOT, damage, &absorb, &resist, m_spellInfo);
 
-    if (target->GetHealth() < damage)
-        damage = uint32(target->GetHealth());
+    if (target->GetMaxHealth() < damage)
+        damage = target->GetMaxHealth();
```

Comment:
Well, GetHealth function will calculate currently amount of HP on target e.g. if target has 500/1000 hp it will calculate amount of 500 and if it's GetMaxHealth function, it will calculate maximal hp of target (1000 HP in this example).